compiler/vm.nim

```diff
@@ -1338,6 +1429,29 @@ proc rawExecute(c: PCtx, start: int, tos: PStackFrame): TFullReg =
       let rb = instr.regBx - wordExcess - 1
       ensureKind(rkNode)
       regs[ra].node = c.globals[rb]
+    of opcLdGlobalDeref:
+      let rb = instr.regBx - wordExcess - 1
+      let node = c.globals[rb]
+      let typ = node.typ
+      doAssert node.kind == nkIntLit, $(node.kind)
+      if typ.kind == tyPtr:
+        ensureKind(rkNode)
+        # use nkPtrLit once this is added
```

Comment:
Maybe ``nkPtrLit`` is indeed the better solution instead of these type based checks.

Reply:
indeed, but I'm leaving this one as a TODO for future PR to avoid introducing regressions unrelated to this PR; the problem predated this PR, for example evalffi used an alias:
```nim
evalffi.nim:47:3:  nkPtrLit = nkIntLit # hopefully we can get rid of this hack soo
```